home/import_content_pages.py

```diff
@@ -84,20 +93,24 @@ def process_rows(self, rows: list["ContentRow"]) -> None:
         # Non-page rows don't have a locale, so we need to remember the last
         # row that does have a locale.
         prev_locale: Locale | None = None
-        for row in rows:
-            if row.is_page_index:
-                if self.locale and row.locale != self.locale.get_display_name():
-                    # This page index isn't for the locale we're importing, so skip it.
-                    continue
-                self.create_content_page_index_from_row(row)
-                prev_locale = self.locale_from_display_name(row.locale)
-            elif row.is_content_page:
-                self.create_shadow_content_page_from_row(row)
-                prev_locale = self.locale_from_display_name(row.locale)
-            elif row.is_variation_message:
-                self.add_variation_to_shadow_content_page_from_row(row, prev_locale)
-            else:
-                self.add_message_to_shadow_content_page_from_row(row, prev_locale)
+        for i, row in enumerate(rows):
+            try:
+                if row.is_page_index:
+                    if self.locale and row.locale != self.locale.get_display_name():
+                        # This page index isn't for the locale we're importing, so skip it.
+                        continue
+                    self.create_content_page_index_from_row(row)
+                    prev_locale = self.locale_from_display_name(row.locale)
+                elif row.is_content_page:
+                    self.create_shadow_content_page_from_row(row)
+                    prev_locale = self.locale_from_display_name(row.locale)
+                elif row.is_variation_message:
+                    self.add_variation_to_shadow_content_page_from_row(row, prev_locale)
+                else:
+                    self.add_message_to_shadow_content_page_from_row(row, prev_locale)
+            except ImportException as e:
+                e.row_num = i + 2  # Rows numbers start at one and have a header row
```

Comment:
You could use `enumerate(rows, 2)` above instead of adding 2 where you use the index.

I think the add is fine in this case, because we're only using the index once, `enumerate()`'s second arg can be useful when the index is referenced in a bunch of different places.